securedrop/tests/test_source.py

```diff
@@ -312,6 +312,38 @@ def test_submit_both(self):
             self.assertIn("Thanks! We received your message and document",
                           resp.data)
 
+    @patch('source_app.main.async_genkey')
+    @patch('source_app.main.get_entropy_estimate')
+    def test_submit_message_with_low_entropy(self, get_entropy_estimate,
+                                             async_genkey):
+        get_entropy_estimate.return_value = 300
```

Comment:
this is the only difference between the two tests. It is problematic in the sense that a regression in the code that would make both tests run the same code path would not raise an error. The logs should be examined to assert a different message is issued, or any other proof that the expected code path has run.